File: aongbucks_server/src/main/java/com/ssafy/cafe/model/service/ProductService.java
package com.ssafy.cafe.model.service;

import java.util.List;
import java.util.Map;

import com.ssafy.cafe.model.dto.Product;

public interface ProductService {
    /**
     * 모든 상품 정보를 반환한다.
     * @return
     */
    List<Product> getProductList();
    
    /**
     * 사용자가 즐겨찾기 설정을 한 상품 정보를 모두 반환
     * @param userId
     * @return
     */
    List<Product> getFavoriteProducts(String userId);
    
    /**
     * backend 관통 과정에서 추가됨
     * 상품의 정보, 판매량, 평점 정보를 함께 반환
     * @param productId
     * @return
     */
    List<Map<String, Object>> selectWithComment(Integer productId, String userId);
    
}

File: AongBucks-Manager/app/src/main/java/com/ssafy/aongbucks_manager/service/ProductService.kt
package com.ssafy.aongbucks_manager.service

import com.ssafy.aongbucks_manager.dto.Product
import com.ssafy.aongbucks_manager.util.RetrofitCallback
import com.ssafy.aongbucks_manager.util.RetrofitUtil
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

private const val TAG = "UserService_싸피"
class ProductService {

    fun getProductList(callback: RetrofitCallback<List<Product>>)  {
        val menuInfoRequest: Call<List<Product>> = RetrofitUtil.productService.getProductList()
        menuInfoRequest.enqueue(object : Callback<List<Product>> {
            override fun onResponse(call: Call<List<Product>>, response: Response<List<Product>>) {
                val res = response.body()
                if(response.code() == 200){
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<List<Product>>, t: Throwable) {
                callback.onError(t)
            }
        })
    }
}